src/vs/workbench/contrib/chat/browser/actions/chatClear.ts

```diff
@@ -4,26 +4,24 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { ServicesAccessor } from 'vs/platform/instantiation/common/instantiation';
-import { IViewsService } from 'vs/workbench/common/views';
-import { type IChatViewPane, type IChatWidget } from 'vs/workbench/contrib/chat/browser/chat';
+import { type IChatWidget } from 'vs/workbench/contrib/chat/browser/chat';
 import { IChatEditorOptions } from 'vs/workbench/contrib/chat/browser/chatEditor';
 import { ChatEditorInput } from 'vs/workbench/contrib/chat/browser/chatEditorInput';
 import { IEditorGroupsService } from 'vs/workbench/services/editor/common/editorGroupsService';
 import { IEditorService } from 'vs/workbench/services/editor/common/editorService';
 
 export async function clearChatSession(accessor: ServicesAccessor, widget: IChatWidget): Promise<void> {
-	if ('viewId' in widget.viewContext) {
-		const viewsService = accessor.get(IViewsService);
-		// This cast is to break a circular dependency- ideally this would not be called directly for `/clear`
-		// from the widget class, but from some contribution.
-		const view = viewsService.getViewWithId(widget.viewContext.viewId);
-		if (!view || !(view as any as IChatViewPane).clear) {
-			return;
-		}
+	const editorService = accessor.get(IEditorService);
+	const editorGroupsService = accessor.get(IEditorGroupsService);
 
-		(view as any as IChatViewPane).clear();
+	const chatEditorInput = editorService.activeEditor;
+	if (chatEditorInput instanceof ChatEditorInput && chatEditorInput.providerId) {
```

Comment:
This doesn't seem right, it's ignoring the `widget` that was passed in, if there is a chat activeEditor (IIRC, that doesn't necessarily mean 'focused')